### PR TITLE
[ACS][Chat] Added File Sharing 

### PR DESCRIPTION
**Pull request by jpeng-ms**
# Description

We are updating the Chat SDK for Java to include file sharing support. No logic changes on SDK side, we only updated the chat client based on latest swagger: 

https://raw.githubusercontent.com/Azure/azure-rest-api-specs/f579ddcd291d55b0a2a32ee81995432458db3da9/specification/communication/data-plane/Chat/readme.md

build will fail until we deploy the new backend API. 

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [X] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [X] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/communication/azure-communication-chat/CHANGELOG.md [hunk 1]
```diff
@@ -1,15 +1,11 @@
...
 ### Features Added
 
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
-
+- Expanded `ChatAttachment` to include new enum `file` to enable communication users to recieve file attachments shared by Teams user in a Teams Interoperbility Chat.
+- Added support for a new communication identifier `MicrosoftTeamsAppIdentifierModel`.
```
Comment (srnagar) [inline sdk/communication/azure-communication-chat/CHANGELOG.md:8]: I don't see this added to the public models. It's only internally added and shouldn't be part of the changelog if it's not publicly added.

Comment (jpeng-ms) [reply sdk/communication/azure-communication-chat/CHANGELOG.md:8]: thanks! will remove it from changelog

File: sdk/communication/azure-communication-chat/pom.xml [hunk 2]
```diff
@@ -93,6 +93,12 @@
       <version>5.9.3</version> <!-- {x-version-update;org.junit.jupiter:junit-jupiter-params;external_dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-netty</artifactId>
+      <version>1.14.2</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
+      <scope>test</scope>
+    </dependency>
```
Comment (srnagar) [inline sdk/communication/azure-communication-chat/pom.xml:101]: This shouldn't be required. `azure-communication-common` and `azure-communication-identity` should have use `current` tag as they both are in the same service group - `<!-- {x-version-update;com.azure:azure-communication-identity;current} -->`

File: sdk/communication/azure-communication-chat/swagger/README.md [hunk 3]
```diff
@@ -35,9 +35,9 @@ To update generated files for chat service, run the following command
 ### Code generation settings
 
 ```yaml
-tag: package-chat-2023-11-07
+tag: package-chat-2024-03-07
```
Comment (srnagar) [inline sdk/communication/azure-communication-chat/swagger/README.md:38]: Add the version of autorest used to ensure everyone uses the same version to regenerate code for this library.
```suggestion
tag: package-chat-2024-03-07
use: '@autorest/java@4.1.25'
```

Comment (jimchou-dev) [reply sdk/communication/azure-communication-chat/swagger/README.md:38]: Regenerated with @autorest/java@4.1.25

File: sdk/communication/azure-communication-chat/pom.xml (hunk 2 above)
Comment (jimchou-dev) [reply sdk/communication/azure-communication-chat/pom.xml:101]: This workaround code to be removed when actual fix is in, tracking the removal task in item 3674953 in ADO

**Review by srnagar — APPROVED** [review]

File: sdk/communication/azure-communication-chat/pom.xml (hunk 2 above)
Comment (jimchou-dev) [reply sdk/communication/azure-communication-chat/pom.xml]: changed scope to compile

File: sdk/communication/azure-communication-chat/swagger/README.md [hunk 4]
```diff
@@ -23,21 +23,22 @@ There is one swagger for Chat management APIs.
...
 
 ### Code generation settings
 
 ```yaml
-tag: package-chat-2023-11-07
+tag: package-chat-2024-03-07
+use: '@autorest/java@4.1.25'
 require:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/731138d0505c02fcee05d7b5a1417fd10ef5efc3/specification/communication/data-plane/Chat/readme.md
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/1720e5aaa4f5ab094789f72589122baa07133439/specification/communication/data-plane/Chat/readme.md
```
Comment (JoshuaLai) [inline sdk/communication/azure-communication-chat/swagger/README.md:41]: This is pointing to the wrong ref. It should be: `https://raw.githubusercontent.com/Azure/azure-rest-api-specs/ac666e993a97fdcda75286b347627c8fd8e7b39d/specification/communication/data-plane/Chat/readme.md`

**Pull request merged**